messaging-client-core/src/main/java/com/inmobi/messaging/consumer/MessageConsumer.java

```diff
@@ -83,4 +83,13 @@ public Message next(long timeout, TimeUnit timeunit)
    * Close and cleanup all resources such as connection, file handles etc.
    */
   public void close();
+
+  /***
+   * Calculates the remaining amount of data for a consumer. Other consumer implementations are
```

Comment:
changed contract, instead of -1. they will throw UnsupportedOperationException now